cocotb/triggers.py

```diff
@@ -152,10 +152,14 @@ def __init__(self):
 
     def unprime(self):
         """Disable a primed trigger, can be re-primed."""
-        if self.cbhdl != 0:
+        # Due to race condition in object global object we rely on may have been
+        # deleted. This will happen only right before exit so just skip over the
+        # action.
+        if self.cbhdl != 0 and simulator is not None:
             simulator.deregister_callback(self.cbhdl)
         self.cbhdl = 0
-        Trigger.unprime(self)
+        if Trigger is not None:
+            Trigger.unprime(self)
```

Comment:
Can this really ever happen?

Reply:
(it being none)

Reply:
Yes, would not have implemented this check; the exceptions at the end of the output given in my first comment are caused by Trigger being None.
Took some time to understand myself how this could happen. It seems that when python cleans up a module at exit he may set class globals to None (here Trigger) before calling __del__ on objects in that module.